cloudslang-runtime/src/main/java/io/cloudslang/lang/runtime/env/RunEnvironment.java

```diff
@@ -121,4 +121,103 @@ public void resetStacks() {
         parentFlowStack = new ParentFlowStack();
     }
 
+    public boolean containsSensitiveData() {
+        return containsSensitiveCallArgument() ||
+                containsSensitiveReturnValues() ||
+                containsSensitiveSystemProperties() ||
+                containsSensitiveContexts();
+    }
+
+    public void decryptSensitiveData() {
+        for (Value value: prepareValuesForEncryptDecrypt()) {
+            if(value.isSensitive()) {
+                ((SensitiveValue)value).decrypt();
+            }
+        }
+    }
+
+    public void encryptSensitiveData() {
+        for (Value value: prepareValuesForEncryptDecrypt()) {
+            if(value.isSensitive()) {
+                ((SensitiveValue)value).encrypt();
+            }
+        }
+    }
+
+    private boolean containsSensitiveCallArgument() {
+        return callArguments != null && containsSensitiveData(callArguments.values());
+    }
+
+    private boolean containsSensitiveReturnValues() {
+        return (returnValues != null) && (returnValues.getOutputs() != null) &&
+                containsSensitiveData(returnValues.getOutputs().values());
+    }
+
+    private boolean containsSensitiveSystemProperties() {
+        return (systemProperties != null) &&
+                containsSensitiveData(Collections2.transform(systemProperties, new Function<SystemProperty, Value>() {
```

Comment:
little concern: maybe we shouldn't make this class dependent of `Jython` because of a `map` functionality